File: test/support/fixtures/document_fixtures/test_code.rb
class PetShop
  ## Represents a pet shop with inventory and sales functionality.

  def initialize(name)
    @name = name
    @inventory = {}
  end

  def add_pet(pet_type, quantity)
    if @inventory.key?(pet_type)
      @inventory[pet_type] += quantity
    else
      @inventory[pet_type] = quantity
    end
  end

  def sell_pet(pet_type, quantity)
    if @inventory.fetch(pet_type, 0) >= quantity
      @inventory[pet_type] -= quantity
      true
    else
      false
    end
  end

  private

  def restock_threshold
    5
  end
end

File: test/support/fixtures/document_fixtures/test_code.php
<?php

class PetShop
{
    /**
     * Represents a pet shop with inventory and sales functionality.
     */
    private $inventory = [];

    public function addPet($petType, $quantity)
    {
        if (isset($this->inventory[$petType])) {
            $this->inventory[$petType] += $quantity;
        } else {
            $this->inventory[$petType] = $quantity;
        }
    }

    public function sellPet($petType, $quantity)
    {
        if (($this->inventory[$petType] ?? 0) >= $quantity) {
            $this->inventory[$petType] -= $quantity;
            return true;
        }

        return false;
    }

    private function restockThreshold()
    {
        return 5;
    }
}
